arraycontext/pytest.py

```diff
@@ -31,7 +31,35 @@
 
 # {{{ pytest integration
 
-def pytest_generate_tests_for_pyopencl_array_context(metafunc):
+import pyopencl as cl
+from pyopencl.tools import _ContextFactory
+
+
+class _PyOpenCLArrayContextFactory(_ContextFactory):
+    def __call__(self):
+        ctx = super().__call__()
+        from arraycontext.impl.pyopencl import PyOpenCLArrayContext
+        return PyOpenCLArrayContext(cl.CommandQueue(ctx))
+
+    def __str__(self):
+        return ("<PyOpenCL array context factory for <pyopencl.Device '%s' on '%s'>"
+                % (self.device.name.strip(),
+                 self.device.platform.name.strip()))
+
+
+class _PytatoArrayContextFactory(_ContextFactory):
+    def __call__(self):
+        ctx = super().__call__()
+        from arraycontext.impl.pytato import PytatoArrayContext
+        return PytatoArrayContext(cl.CommandQueue(ctx))
+
+    def __str__(self):
+        return ("<Pytato array context factory for <pyopencl.Device '%s' on '%s'>"
+                % (self.device.name.strip(),
+                 self.device.platform.name.strip()))
+
+
+def pytest_generate_tests_for_array_contexts(metafunc, use_pytato=True) -> None:
```

Comment:
This should take a list of array context classes.

Reply:
Done in https://github.com/inducer/arraycontext/pull/14/commits/14bf6045407ba614f88255d860a5da419c7b8018